### ISSUE TITLE
?: operator in where statement

### ISSUE THREAD

**Issue by UmmonKRS** [linq2db/linq2db#4052]
Hello!

Can I use the ternary conditional operator in where statement?

```cs
            // works fine
            var ev = data.Street != null ? data.Street.Trim().ToLower() : string.Empty;
            // NRE
            var query1 = (from c in db.Claims
                          where c.DistrictId == data.DistrictId
                             && (c.Street != null ? c.Street.Trim().ToLower() : string.Empty) == (data.Street != null ? data.Street.Trim().ToLower() : string.Empty)
                          select c).ToList();
```
where "data" is fake Claim object, which is a set of conditions to search in db, so field can be null.

How can I do it in a proper way?

### Environment details

`Linq To DB` version: 3.0.1
Database (with version): PostgreSQL
.NET Version: FW4.8

**Comment by sdanyliv** [comment]
Try to update `linq2db` first. Probably in two years issue is already fixed.

**Comment by UmmonKRS** [comment]
Ok, I've made a stub project, FW4.8, linq2db 5.1.1
Trying to generate model over my db with T4, it failed with:
System.InvalidOperationException: Connection string is not provided.
on
LinqToDB.DataProvider.PostgreSQL.PostgreSQLTools.CreateDataConnection("Server=localhost;Port=5432;Database=npsub;User Id=postgres;Password=...;Pooling=true;MinPoolSize=10;MaxPoolSize=100;");
(even try to put just this line in static Main, no T4)
Same single line of code in old project works fine.
Same code in new stub project, FW4.8, linq2db 3.0.1 works fine.

What is wrong here?..

**Comment by MaceWindu** [comment]
#4045

**Comment by sdanyliv** [comment]
@UmmonKRS, do not scaffold db right now. Just test how it works with current version.

**Comment by UmmonKRS** [comment]
@MaceWindu Thank you for information!

@sdanyliv when I trying to add generated (old project, linq2db v3) file to a new project I get an error on Association attribute

[Association(ThisKey = "DistrictId", OtherKey = "DistrictId", CanBeNull = false, Relationship = Relationship.ManyToOne, KeyName = "registry_fk_district", BackReferenceName = "Registryfks")]

The name 'Relationship' does not exist in the current context

It seems something had changed in attribute system.

**Comment by MaceWindu** [comment]
`Relationship`, `BackReferenceName` and `KeyName` were removed and not used.

**Comment by MaceWindu** [comment]
Made some tests - issue was fixed starting from release 3.4.4
